src/main/java/spoon/support/compiler/jdt/ReferenceBuilder.java

```diff
@@ -542,7 +544,9 @@ private boolean isCorrectTypeReference(TypeReference ref) {
 		}
 		if (!res.toString().replace(", ?", ",?").endsWith(CharOperation.toString(ref.getParameterizedTypeName()))) {
 			// verify that we did not match a class that have the same name in a different package
-			return this.jdtTreeBuilder.getFactory().Type().createReference(CharOperation.toString(ref.getParameterizedTypeName()));
+			CtTypeReference result = this.jdtTreeBuilder.getFactory().Type().createReference(res.getQualifiedName());
+			result.setActualTypeArguments(res.getActualTypeArguments());
+			return result;
```

Comment:
@tdurieux could you review this piece of code: you introduce the condition in #1480 to fix another bug

Reply:
Your patch is invalid, it brokes my test (and my test is good)

Reply:
the bug is in getQualifiedName of jdt in my testcase

Reply:
> Your patch is invalid, it brokes my test (and my test is good)

Yeah, I saw that. Still working on it, I come back to you when tests are passing, I was a bit too confident on the first trial ;)